### DISCUSSION TITLE
Dealing with native DATE fields (without time) on MySQL

### DISCUSSION THREAD

**Discussion by lucashfreitas** [prisma/prisma#13773] [Q&A]
I have a **DATE** field in the database - without time.

``` 
 date_to_process             DateTime @db.Date()
```

I am trying to run queries in this field but it's not returning any rows as Prisma is sending a time info string to the MySQL query, something like this ```where date_to_process ='2022-06-12 12:52:28.043 UTC```

I can't pass a string like `2022-06-13` as the Prisma client will throw an error as it expects a javascript date object. If I try to give a date object from today Prisma will send a timezone string to the query resulting in no rows returned as mentioned above. 

Mysql allow us to use string on date filter natively:

``` MySQL
SELECT * FROM Order
where date_to_process ='2022-06-12' -- that works fine on MySQL.
```

Could Prisma identify that we are dealing with a DATE type only and "strip out the time part when we use a javascript Date Object?". I believe we could try a workaround to reset the time portion of the Date Object but as Prisma knows that is a ```@db.Date()``` that could be done - or perhaps allow us to use string as we can do that in native mysql.

This is the query I am running:

``` typescript

  const items = await db.subscriptionOrder.findMany({
    
    where: {
      date_to_process: new Date(),
    },
    
```

Thanks

**Comment by janpio** [comment]
Is this the real query MySQL is executing? Or is that an assumption?
What happens if you send the JS date for `00:00:00.000 UTC`?

**Reply by lucashfreitas** [reply]
This is the parameter outputed by the SQL string: 

I was trying to see what query prisma is sending to understand why no results are being returned and the parameter that is being logged has the 00:00:00.000 UTC string. I did not find much content related to date (without time) filtering, so perhaps its better for me to raise an issue and a demo unless this is a known issue? 

Thank you

**Reply by janpio** [reply]
You send in a `Date()`, so that that is what Prisma Client is sending to the engine and then using in the query.
For a `DateTime @db.Date` type field you should be able to send in a JS datetime with the date part set to 0 values for this to work. If that does not work, a bug report issue indeed would be good.

**Reply by lucashfreitas** [reply]
@janpio - I can confirm that if we send a javascript date object with a time different than 0 the query won't work. Prisma also doesn't support a string, e.g - `2022-06-25`  - if we try it will throw an error - I found it a bit odd as MySQL natively support filtering date column using a string notation.

```
Got invalid value '2022-06-22' on prisma.findManySubscriptionOrderSchedule. Provided String, expected DateTimeFilter or DateTim
```
 I am away now but will try to create a reproduction asap. 

Thanks

**Reply by lucashfreitas** [reply]
@janpio 

I can also confirm that despite the field being a `@db.Date()`, Prisma is sending the whole date with timezone string while filtering it to MySQL - due to this the database doesn't return results. I have enabled logs both on prisma and on MySQL engine level and this is the value that prisma is sending to MySQL:

<img width="449" alt="image" src="https://user-images.githubusercontent.com/11416392/175774370-51278b75-97fd-4585-92f0-3194756311f4.png">

**Reply by pantharshit00** [reply]
Hello @lucashfreitas,

Did you get a chance to create a reproduction? If this is still relevant, it would be best if you can report this as a bug report: https://github.com/prisma/prisma/issues/new?assignees=&labels=kind/bug&template=bug_report.yml

**Reply by danieltan007** [reply]
i have same problem, is there any update?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
